craft_providers/multipass/multipass.py

```diff
@@ -386,3 +387,65 @@ def umount(self, *, mount: str) -> None:
                 brief=f"Failed to unmount {mount!r}.",
                 details=errors.details_from_called_process_error(error),
             ) from error
+
+    def wait_until_ready(
+        self, *, retry_wait: float = 0.25, timeout: Optional[float] = None
+    ):
+        """Wait until Multipass is ready (upon install/startup).
+
+        :param retry_wait: Time to sleep between retries.
+        :param timeout: Optional time.time() stamp to timeout after.
+        """
+        while timeout is None or time.time() < timeout:
+            multipass_version, multipassd_version = self.version()
+
+            if multipassd_version is not None:
+                return (multipass_version, multipassd_version)
```

Comment:
It's not obvious (not documented) that `wait_until_ready` returns a tuple of versions. Perhaps we could leave that to the `version()` call?

Reply:
I fixed the docs and return annotation.

It's really just to avoid an unnecessary call to version since it will be available. It's almost a half second in some cases.